src/ui/public/visualize/visualize.js

```diff
@@ -144,7 +144,12 @@ uiModules
         }
 
         if (oldVis) $scope.renderbot = null;
-        if (vis) $scope.renderbot = vis.type.createRenderbot(vis, $visEl, $scope.uiState);
+        if (vis) {
+          vis.on('renderComplete', () => {
+            $scope.$emit('renderComplete');
+          });
```

Comment:
Yeah, that's fine with me. I just need something to listen to, I don't care what that something is.